### ISSUE TITLE
Tac/Scan cocktail mode not flipping

### ISSUE THREAD

**Issue by rdagger** [libretro/mame2003-plus-libretro#1585]
Despite setting the dip switch in Tac/Scan to Cocktail mode, the screen doesn't flip for player 2.  The [game flyer](https://flyers.arcade-museum.com/?page=thumbs&db=videodb&id=1163) shows both standup and cocktail cabinets.

**Comment by arcadez2003** [comment]
It's listed as a bug on the arcade database although sometimes these comments are outta date

"The cocktail DIP has no effect and there are no separate controls for player 2. Frotz (ID 00260)"

Does it flip correctly using current MAME.??

**Comment by rdagger** [comment]
Current MAME doesn't flip either.  It's strange because the [owner's manual](https://archive.org/details/arcademanual_tac-scan/page/n9) specifies a DIP switch for cocktail mode.

**Comment by arcadez2003** [comment]
> Current MAME doesn't flip either. It's strange because the [owner's manual](https://archive.org/details/arcademanual_tac-scan/page/n9) specifies a DIP switch for cocktail mode.

Yeah i believe ya it's just that option needs to be coded in both the driver and the video code, i guess no one on the
MAME dev team has gotten around to it yet.

**Comment by mahoneyt944** [comment]
I saw that all these games in the driver have cocktail dips. Apparently these dips are non standard, accordingly to the driver documentation. Would have to look at the drawing bits to see if it just needs flipped, could possibly fix a handful of these if so.

**Comment by mahoneyt944** [comment]
I've been looking over the code and trying to research this a little bit. I can't seem to find so much as an image of an actual tac/scan cocktail table. Also noted in mame's source, the cocktail switch is not mentioned in the owners manual. It's quite possible this switch was just never used officially and therefore isn't meant to function? Of course this is just speculation, but normally there's at least some information to go by.

I think it's doable to hack in flipping capability but I'd like to verify that this functionality actually existed on real cocktail tables first. Seems like it maybe just an unused switch?

**Comment by rdagger** [comment]
The flyer shows a cocktail table and [klov ](https://www.arcade-museum.com/Videogame/tac-scan) indicates a cocktail cabinet.
![tacscan cocktail](https://github.com/libretro/mame2003-plus-libretro/assets/106355/4796ff10-3033-467d-96ff-b92b5872580b)

**Comment by rdagger** [comment]
Page 8 of the [service manual](http://pdf.textfiles.com/manuals/ARCADE/S-Z/Tac%20Scan%20[Owner%27s]%20[English].pdf) shows that switch #1 controls the cocktail mode.  Page 117 (left side) shows the schematic for switch #1.

**Comment by mahoneyt944** [comment]
I hooked up a test flip for these sega games, https://github.com/libretro/mame2003-plus-libretro/compare/master...Sega

Just need to figure out what bit is suppose to control the flipping. Currently I have it set to flip with the dip switch for testing.
